app/src/main/java/com/therap/amin/currencyconverter/Fragments/CurrencyConversionFragment.java

```diff
@@ -72,8 +149,36 @@ public void afterTextChanged(Editable s) {
             }
         });
 
-        tvInputCurrency.setText(sharedPreferences.getString(Constants.INPUT_CURRENCY_KEY, "Not Set"));
-        tvOutputCurrency.setText(sharedPreferences.getString(Constants.OUTPUT_CURRENCY_KEY, "Not Set"));
         return view;
     }
+
+    public void updateUI () {
+        String from = fromCurrencySpinner.getSelectedItem().toString();
+        String to = toCurrencySpinner.getSelectedItem().toString();
+
+        fileProcessor = new FileProcessor(getActivity());
+        Double conversionRate = null;
+        if (selectedSource.equals("own")) {
+            String valueFromPreference = sharedPreferences.getString(from + to, null);
+            if (valueFromPreference != null) conversionRate = Double.parseDouble(valueFromPreference);
+        } else if (selectedSource.equals("web") && fileProcessor.calculateConversionRate(from,to) != -1) {
+            conversionRate = fileProcessor.calculateConversionRate(from,to);
+        }
+        if (conversionRate != null) {
+            tvSavedCurrencyRate.setText(String.format("1 %s = %.2f %s",from,conversionRate,to));
+            if (!etInputCurrencyValue.getText().toString().equals("")) {
+                conversionRate *= Double.parseDouble(etInputCurrencyValue.getText().toString());
+                tvOutputCurrencyValue.setText(String.format("%.2f", conversionRate));
+            }
+        } else {
+            tvSavedCurrencyRate.setText(R.string.not_available);
+        }
+    }
+
+    @Override
+    public void onDestroy() {
```

Comment:
Don't use onDestryo for saving state. There is no guarantee that it will call. Use onSaveInstanceState() or onPause() method. Read carefully documentation at:
https://developer.android.com/guide/components/activities.html#SavingActivityState